### PR TITLE
Add `configure_columns` method to RichProgressBar

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Added `configure_columns` method to RichProgressBar for Users to customize the columns

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/callbacks/test_rich_progress_bar.py [hunk 1]
```diff
@@ -141,3 +141,24 @@ def on_train_start(self) -> None:
...
+
+    progress_bar = CustomRichProgressBar()
+
+    model = BoringModel()
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, callbacks=progress_bar)
+
+    trainer.fit(model)
+
+    assert progress_bar.progress.columns[0] == custom_column
+    assert len(progress_bar.progress.columns) == 1
```
Comment (SeanNaren) [inline tests/callbacks/test_rich_progress_bar.py:164]: Not sure how much work it is, but we could convert the need to run this end to end into a unit test potentially? might be able to just instantiate CustomRichProgressBar and call `configure_columns` passing a trainer/model object and assert the progress bar contains the column?

Albeit this doesn't test the full end to end logic ensuring that we call this in the right place when trainer.fit is called

Comment (kaushikb11) [reply tests/callbacks/test_rich_progress_bar.py]: Good point, @SeanNaren Updated!

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !

File: CHANGELOG.md [hunk 2]
```diff
@@ -124,6 +124,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `auto_device_count` method to `Accelerators` ([#10222](https://github.com/PyTorchLightning/pytorch-lightning/pull/10222))
 - Added support for `devices="auto"` ([#10264](https://github.com/PyTorchLightning/pytorch-lightning/pull/10264))
 - Added a `filename` argument in `ModelCheckpoint.format_checkpoint_name` ([#9818](https://github.com/PyTorchLightning/pytorch-lightning/pull/9818))
+- Added `configure_columns` method to RichProgressBar ([#10288](https://github.com/PyTorchLightning/pytorch-lightning/pull/10288))
```
Comment (rohitgr7) [inline CHANGELOG.md:127]:
```suggestion
- Added `configure_columns` method to `RichProgressBar` ([#10288](https://github.com/PyTorchLightning/pytorch-lightning/pull/10288))
```

File: pytorch_lightning/callbacks/progress/rich_progress.py [hunk 3]
```diff
@@ -435,3 +426,17 @@ def main_progress_bar(self) -> Task:
     @property
     def test_progress_bar(self) -> Task:
         return self.progress.tasks[self.test_progress_bar_id]
+
+    def configure_columns(self, trainer, pl_module) -> list:
```
Comment (rohitgr7) [inline pytorch_lightning/callbacks/progress/rich_progress.py:430]: no mypy?

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/callbacks/progress/rich_progress.py (hunk 3 above)
Comment (kaushikb11) [reply pytorch_lightning/callbacks/progress/rich_progress.py:430]: Could follow up with that
